channels/offline-channel-js/src/OfflineChannel.ts

```diff
@@ -0,0 +1,604 @@
+
+import dynamicProto from "@microsoft/dynamicproto-js";
+import {
+    BreezeChannelIdentifier, DEFAULT_BREEZE_ENDPOINT, DEFAULT_BREEZE_PATH, EventPersistence, IConfig, IOfflineListener,
+    createOfflineListener
+} from "@microsoft/applicationinsights-common";
+import {
+    BaseTelemetryPlugin, IAppInsightsCore, IChannelControls, IConfigDefaults, IConfiguration, IDiagnosticLogger, IInternalOfflineSerializer,
+    INotificationListener, IPayloadData, IPlugin, IProcessTelemetryContext, IProcessTelemetryUnloadContext, ITelemetryItem,
+    ITelemetryPluginChain, ITelemetryUnloadState, IXHROverride, SendRequestReason, arrForEach, createProcessTelemetryContext,
+    createUniqueNamespace, dateNow, mergeEvtNamespace, onConfigChange, runTargetUnload
+} from "@microsoft/applicationinsights-core-js";
+import { IPromise, ITaskScheduler, createAsyncPromise, createTaskScheduler } from "@nevware21/ts-async";
+import { ITimerHandler, isFunction, objDeepFreeze, scheduleTimeout } from "@nevware21/ts-utils";
+import { isGreaterThanZero } from "./Helpers/Utils";
+import { InMemoryBatch } from "./InMemoryBatch";
+import { IPostTransmissionTelemetryItem } from "./Interfaces/IInMemoryBatch";
+import {
+    IOfflineBatchHandler, OfflineBatchCallback, OfflineBatchStoreCallback, eBatchSendStatus, eBatchStoreStatus
+} from "./Interfaces/IOfflineBatch";
+import {
+    ILocalStorageConfiguration, ILocalStorageProviderContext, IOfflineSenderConfig, IStorageTelemetryItem, eStorageProviders
+} from "./Interfaces/IOfflineProvider";
+import { OfflineBatchHandler } from "./OfflineBatchHandler";
+import { isValidPersistenceLevel } from "./Providers/IndexDbProvider";
+import { Sender } from "./Sender";
+
+const version = "#version#";
+const DefaultOfflineIdentifier = "OfflineChannel";
+const DefaultBatchInterval = 15000;
+const DefaultInMemoMaxTime = 15000;
+const PostChannelIdentifier = "PostChannel";
+
+
+interface IUrlLocalStorageConfig {
+    iKey: string;
```

Comment:
We shouldn't need this (unless we want to support different persistence cache level for different iKeys -- which would be possible for 1DS with the Shared Manager, but it's part of the general design